internal/provider/providers/spaceship/createrecord.go

```diff
@@ -0,0 +1,70 @@
+package spaceship
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/provider/errors"
+	"github.com/qdm12/ddns-updater/internal/provider/utils"
+)
+
+func (p *Provider) createRecord(ctx context.Context, client *http.Client,
+	recordType, address string) error {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "spaceship.dev",
+		Path:   fmt.Sprintf("/api/v1/dns/records/%s", p.domain),
+	}
+
+	createData := struct {
+		Force bool `json:"force"`
+		Items []struct {
+			Type    string `json:"type"`
+			Name    string `json:"name"`
+			Address string `json:"address"`
+			TTL     int    `json:"ttl"`
+		} `json:"items"`
+	}{
+		Force: true,
+		Items: []struct {
+			Type    string `json:"type"`
+			Name    string `json:"name"`
+			Address string `json:"address"`
+			TTL     int    `json:"ttl"`
+		}{{
+			Type:    recordType,
+			Name:    p.owner,
+			Address: address,
+			TTL:     3600,
```

Comment:
You might need to define this before as a constant with the linter mnd in place:

```go
const defaultTTL = 3600
```